src/Microsoft.Extensions.Hosting.Abstractions/IHostBuilder.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.Extensions.Hosting
+{
+    /// <summary>
+    /// A program initialization abstraction.
+    /// </summary>
+    public interface IHostBuilder
+    {
+        /// <summary>
+        /// A central location for sharing state between components during the host building process.
+        /// </summary>
+        IDictionary<object, object> Properties { get; }
```

Comment:
Is this something we want to add to WebHostBuilder as well (in the next major release given that this is a breaking change to the interface)?

Reply:
Nah.